samples/watchdog/README.rst

```diff
@@ -0,0 +1,101 @@
+.. _watchdog_sample:
+
+Watchdog Timer
+##############
+
+.. contents::
+   :local:
+   :depth: 2
+
+The Watchdog Timer sample demonstrates how to configure a PMIC TIMER periphery in the Watchdog Timer mode with the use of npmx drivers.
```

Comment:
Perhaps whole directory could be renamed to `samples/timer_watchdog` so it's aligned with `timer_wakeup`?